cmd/akamai-purger/main.go

```diff
@@ -150,19 +169,21 @@ func main() {
 	cmd.FailOnError(err, "Unable to setup Akamai purger gRPC server")
 	akamaipb.RegisterAkamaiPurgerServer(grpcSrv, &ap)
 
-	go cmd.CatchSignals(logger, grpcSrv.GracefulStop)
+	go cmd.CatchSignals(logger, func() {
+		grpcSrv.GracefulStop()
+		// Stop the ticker and signal that we want to shutdown by writing to the
+		// stop channel. We wait 15 seconds for any remaining URLs to be emptied
+		// from the current queue, if we pass that deadline we exit early.
+		ticker.Stop()
+		stop <- true
+		select {
+		case <-time.After(time.Second * 15):
+			cmd.Fail("Timed out waiting for purger to finish work")
+		case <-stopped:
+		}
+	})
 
 	err = cmd.FilterShutdownErrors(grpcSrv.Serve(l))
 	cmd.FailOnError(err, "Akamai purger gRPC service failed")
-
-	// Stop the ticker and signal that we want to shutdown by writing to the
-	// stop channel. We wait 15 seconds for any remaining URLs to be emptied
-	// from the current queue, if we pass that deadline we exit early.
-	ticker.Stop()
-	stop <- true
-	select {
-	case <-time.After(time.Second * 15):
-		cmd.Fail("Timed out waiting for purger to finish work")
-	case <-stopped:
-	}
+	select {}
```

Comment:
Could you add a comment for why this is needed, it's a little confusing at first sight.